GOVERNANCE.md

```diff
@@ -1,131 +1,187 @@
 
 ### Committers
-If you make regular contribution to OpenRefine, you will most likely become a Committer.
+If you make regular contributions to OpenRefine, you will most likely become a Committer.
 
-Committers have earned enough trust from the community to review and merge pull requests.
+Committers have earned enough trust from the community to
 
-Therefore Committers:
-- Help contributors via the [developer discussion list](https://forum.openrefine.org/c/dev/8).
 - Triage issues, pull requests and [projects](https://github.com/OpenRefine/OpenRefine/projects)
 - Have direct access to the code base to create new branches
-- Organize the wiki
-- Review and merge pull requests submitted by contributors
+- Organize the GitHub wiki
+- Review pull requests submitted by contributors
+- Run the CI pipeline on your PRs automatically
 - Are part of the OpenRefine organization and have the OpenRefine badge on their GitHub profile
 
-#### How to become a Committer?  
+**How to become a Committer?**: Committers are voting-binding members; they select and elect new members through a voting process. Nominations should be submitted to the [developer discussion list](https://forum.openrefine.org/c/dev/8), where contributors can nominate themselves or other contributors. See the Member Election section below for more details.
+
+**Current list of Committers**: The list is available here: https://github.com/orgs/OpenRefine/people.
 
-Be a contributor and be nominated as a Committer. Current Committers select and elect new Committers. You may nominate yourself. Nomination should be sent to the [developer discussion list](https://forum.openrefine.org/c/dev/8)
+### Core Developer Group
+Also referred to as Core Dev. The Core Developer Group are trusted developers who
+- Merge pull requests
+- Publish releases
+- Review security vulnerability reports
+- Provide long-term project vision with the community
+- Are responsible for the technical direction of the project
+- Are part of the Admin team for the project on GitHub
 
-#### Current list of Committers
-The list is available here: https://github.com/orgs/OpenRefine/people.
+**How to join the Core Dev Group?**. To be eligible to serve on the Core Developers, individuals must first be Committers of the project. Core Dev Group members are voting-binding members; they select and elect new members through a voting process. Nominations should be submitted to the [developer discussion list](https://forum.openrefine.org/c/dev/8), where committers can nominate themselves or other committers. See the Member Election section below for more details.
 
-### Release Manager
+**Current list of Core Developers:**
+- Tom Morris [@tfmorris](https://github.com/tfmorris) and on the [forum](https://forum.openrefine.org/u/tfmorris/summary)
+- Antonin Delpeuch until March 2025: [@wetneb](https://github.com/wetneb) and on the [forum](https://forum.openrefine.org/u/antonin_d/summary).
```

Comment:
The date for Antonin has already passed. Is it still valid? 

We need to figure out a way to bootstrap this group. One, or even two, members is too few. @Abbe98 his provided useful technical feedback in the past. Perhaps he could be a candidate, if willing.